sway-lib-std/src/assert.sw

```diff
@@ -10,3 +11,17 @@ pub fn assert(a: bool) {
         ()
     }
 }
+
+/// A wrapper for `assert` that allows logging a custom value `v` if condition `c` is not true.
+/// This will then revert with the value `42`, which indicates that you should look at the previous logd receipt for further debugging clues. Note that the SDK will not currently decode this logd receipt for you.
```

Comment:
I'm not attached to the value `42` here as a signal value, but we do need to use something.

Reply:
Is it worth making some special constants, only used internally within the stdlib but still semi-documented so when we see 42 we know where to look?